compiler/luci/service/src/CircleShapeInferenceHelper.h

```diff
@@ -87,6 +87,61 @@ class TensorShapeExpander final
   const loco::TensorShape _shape;
 };
 
+template <class CIRCLENODE>
```

Comment:
It would be better to move to .cpp. this method seems long to exist in header.

Reply:
> It would be better to move to .cpp. this method seems long to exist in header.

Since this function is template function, the implementation should be in header.
This function is used by multiple classes (CirclePad, CirclePadV2, CircleMirrorPad), so I decided to place it in Helper.h
Would it be better to remove template?

Reply:
> Would it be better to remove template?

I think that will make duplicate codes.
Let's move to separate header file for Pad Ops.

Reply:
> I think that will make duplicate codes. Let's move to separate header file for Pad Ops.

How about luci/Service/src/CircleShapeInferenceHelperForPadOps.h ?